src/main/java/org/jenkins/tools/test/model/plugin_metadata/PluginMetadataHooks.java

```diff
@@ -0,0 +1,99 @@
+package org.jenkins.tools.test.model.plugin_metadata;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+import java.util.jar.JarInputStream;
+import java.util.jar.Manifest;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import org.apache.maven.model.Model;
+import org.jenkins.tools.test.exception.MetadataExtractionException;
+import org.jenkins.tools.test.exception.PluginCompatibilityTesterException;
+import org.jenkins.tools.test.exception.PluginSourcesUnavailableException;
+import org.jenkins.tools.test.exception.WrappedPluginCompatibilityException;
+import org.jenkins.tools.test.model.hook.HookOrderComparator;
+import org.jenkins.tools.test.util.ModelReader;
+
+public class PluginMetadataHooks {
+
+    private static final Logger LOGGER = Logger.getLogger(PluginMetadataHooks.class.getName());
+
+    public static List<PluginMetadataExtractor> loadExtractors(Set<File> externalJars) {
+        ClassLoader cl = setupExternalClassLoaders(externalJars);
+        List<PluginMetadataExtractor> extractors = ServiceLoader.load(PluginMetadataExtractor.class, cl).stream()
+                .map(e -> e.get())
+                .sorted(new HookOrderComparator())
+                .collect(Collectors.toList());
+        return extractors;
+    }
+
+    private static ClassLoader setupExternalClassLoaders(Set<File> externalJars) {
+        ClassLoader base = PluginMetadataHooks.class.getClassLoader();
+        if (externalJars.isEmpty()) {
+            return base;
+        }
+        List<URL> urls = new ArrayList<>();
+        for (File jar : externalJars) {
+            try {
+                urls.add(jar.toURI().toURL());
+            } catch (MalformedURLException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        return new URLClassLoader(urls.toArray(new URL[0]), base);
+    }
```

Comment:
This entails a bit of duplication with `PluginCompatTesterHooks`; can it be de-duped?

Reply:
it can.  I saw this whole class as temporary to be removed once everything had adjusted to the new parent pom, but can extract the commonality for sure.